.github/workflows/msrv.yml

```diff
@@ -10,61 +10,49 @@ jobs:
       matrix:
         rust:
           - 1.70.0
-          - stable
-          - beta
-          - nightly
 
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4.1.1
 
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
```

Comment:
Move this after you have installed your toolchain.

If you are only going to use the projects MSRV, you're not using a matrix anymore for this job and could adjust that.